react/ProductCustomizer.js

```diff
@@ -1,54 +1,180 @@
+    enableChangeToppings: PropTypes.bool,
+  }
+
+  static defaultProps = {
+    enableChangeToppings: false,
+  }
+
+  state = {
+    total: 0,
+    variations: [],
+    isModalOpen: false,
+    selectedVariations: [],
+    selectionsValidated: false,
+  }
+
+  /**
+  * constructor
+  * Initilize the parent class contructor and parse the data from Schema
+  * @return void
+  */
   constructor() {
     super()
-    this.state = { isModalOpen: false }
-    this.handleOpenModal = this.handleOpenModal.bind(this)
-    this.handleCloseModal = this.handleCloseModal.bind(this)
+
+    const service = new ProductCustomizerService(schema)
+
+    this.state.variations = service.serializeData()
   }
 
-  handleOpenModal() {
+  /**
+  * onHandleRemovePreviousSelectedVariation
+  * Remove the current value from variations list and adds the new one.
+  * @param object variationToRemove
+  * @return bool|void
+  */
+  onHandleRemovePreviousSelectedVariation = (variationToRemove) => {
+    const selectedVariations = this.state.selectedVariations
+
+    const updatedSelectedArray = selectedVariations.filter((currentVariation) => {
+      return variationToRemove === undefined || (variationToRemove.Id !== currentVariation.Id)
+    })
+
+    this.setState({ selectedVariations: updatedSelectedArray })
+  }
+
+  /**
+  * onHandleAddSelectedVariations
+  * Add a new selected variation to array then updates total amount.
+  * @param object variation
+  * @return void
+  */
+  onHandleAddAndCalculateSelectedVariations = (variation) => {
+    const selectedVariations = this.state.selectedVariations
+
+    selectedVariations.push(variation)
+    this.setState({ selectedVariations: selectedVariations })
+
+    this.calculateTotal()
+  }
+
+  /**
+  * calculateTotal
+  * Calculate total amount of selected variations
+  * @return void
+  */
+  calculateTotal = () => {
+    let currentTotal = 0
+    const selectedVariations = this.state.selectedVariations
+
+    if (selectedVariations.length === 0) {
+      return 0
+    }
+
+    for (const selectedVariation of selectedVariations) {
+      currentTotal += selectedVariation.price
+    }
+
+    this.setState({ total: currentTotal })
+  }
+
+  /**
+  * handleOpenModal
+  * Show the product modal
+  * @return void
+  */
+  handleOpenModal = () => {
     this.setState({ isModalOpen: true })
   }
 
-  handleCloseModal() {
+  /**
+  * handleCloseModal
+  * Close the product modal
+  * @return void
+  */
+  handleCloseModal = () => {
     this.setState({ isModalOpen: false })
   }
 
+  handleSubmit = e => {
+    console.log(this.state.selectedVariations)
+    e.preventDefault()
+
+    // Checl Lists validations
+    // Add product to cart
+  }
+
   render() {
+    const {
+      variations,
+      selectionsValidated,
+      onHandleUpdateAmount,
+    } = this.state
+
+    const { enableChangeToppings } = this.props
+
     return (
       <div>
         <Button onClick={this.handleOpenModal}>Customize your product</Button>
```

Comment:
Remember to internationalize the entire app.

Reply:
Some texts will came from product details, this one is one of these